### PR TITLE
Update UserAdmin config to help maintenance and support on user accounts

### PR DESCRIPTION
**Pull request by amatissart**
Among the features brought by the standard `UserAdmin`, it provides a search bar to find users by email address.

This will help with the manual operations that are required by #232.

### CODE REVIEW

File: backend/core/admin.py [hunk 1]
```diff
@@ -25,8 +28,27 @@ class DegreeAdmin(admin.ModelAdmin):
...
+    ordering = ['-date_joined']
+
+    def get_fieldsets(self, request, obj) -> List[Tuple]:
+        fieldsets = super().get_fieldsets(request, obj=obj)
+        if not obj:
+            return fieldsets
+
+        # For existing user, show all fields from our custom User model that
+        # are not already present on the page, in a separate fieldsets.
+        existing_fields = flatten_fieldsets(fieldsets)
```
Comment (lfaucon) [inline backend/core/admin.py:41]: Slightly more performant by calling `set(...)` here.

File: backend/core/admin.py [hunk 2]
```diff
@@ -25,8 +28,27 @@ class DegreeAdmin(admin.ModelAdmin):
...
 @admin.register(User)
-class UserAdmin(admin.ModelAdmin):
-    pass
+class UserAdmin(DjangoUserAdmin):
+    ordering = ['-date_joined']
+
+    def get_fieldsets(self, request, obj) -> List[Tuple]:
+        fieldsets = super().get_fieldsets(request, obj=obj)
+        if not obj:
+            return fieldsets
```
Comment (lfaucon) [inline backend/core/admin.py:37]: This doesn't seem necessary. We might prefer it to raise an error if it happens?

**Review by lfaucon — APPROVED** [review]
The user admin got a lot nicer! Great. Approved with suggestions

File: backend/core/admin.py (hunk 1 above)
Comment (amatissart) [reply backend/core/admin.py]: :+1:
Done.

File: backend/core/admin.py (hunk 2 above)
Comment (amatissart) [reply backend/core/admin.py:43]: This case is actually necessary, in order to initialize the admin form used to create a new user.  
(Via the "Add User" button, or directly on http://localhost:8000/admin/core/user/add/ )

**Pull request merged**